internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -282,7 +282,12 @@ func (h *Handler) CreateShare(w http.ResponseWriter, r *http.Request) {
 	reqRole, reqPermissions := r.FormValue("role"), r.FormValue("permissions")
 	switch shareType {
 	case int(conversions.ShareTypeUser), int(conversions.ShareTypeGroup):
-		// user collaborations default to Manager (=all permissions)
+		// NOTE: clients tend to send "31" as permissions but they mean "15". "31" is no longer allowed as we have no resharing any more.
+		if reqRole == "" && reqPermissions == "31" {
+			reqPermissions = "15"
+		}
+
+		// user collaborations default to Manager (=all permissions). This will result in a BadRequest. Sane Default.
```

Comment:
as suggestion

```
if reqRole == "" && permissions > 15 {
		permissions -= 15
	}
```

Reply:
I forwarded the question. I'd go for the simplest fix possible. This is a hack on a deprecated API anyways.

Reply:
> as suggestion
>```
>if reqRole == "" && permissions > 16 {
>		permissions -= 16
>}
>```
That doesn't work unfortunately. `permissions` is a string. It would need to be an int to support substraction.